rfc-013-use-docker-to-deploy-and-run-applications-in-containers.md

```diff
@@ -0,0 +1,76 @@
+# Use Docker to deploy and run our applications in containers
+
+## Summary
+
+All of the applications we build are tested and run using Docker containers.
+
+This includes production environments as well as any staging environments and
+continuous integration services.
+
+The decision on whether we MUST use containers in development is not
+included in the scope of this RFC.
+
+## Problem
+
+To date there has been no formal decision to use or not use containers for
+hosting the applications we build. The result of this is a mixed ecosystem that
+is continuously growing.
+
+We currently host our work using numerous services and platforms:
+
+- Govpress
+- AWS
+- Azure
+- Heroku
+- Dalmatian
+- Government Platform as a Service (GPaaS)
+
+We cannot reduce the amount of services we use as doing so would reduce the
+flexibility we offer our clients. Clients sometimes require we use their tooling
+and platforms ensuring all their services can be managed centrally. This makes
+their services easier to operate as they move into live, it also allows dxw to
+more easily transition away from.
+
+Most applications are already running with containers but some are not.
+
+There are at least 2 category of problem. The missed advantages of not
+containerising our applications and the problems that come from being
+inconsistent in our approach.
+
+There is a common set of problems for software teams that containerisation can
+help us with:
+
+- managing dependencies
+- updating software
+- debugging issues
+- configuring host machines
+- testing and releasing code
+- rolling back
+
+The problems of being inconsistent include:
+
+- common tasks require different processes to achieve the same output. These
+require time to learn and is true whether you are joining the product team or
+are part of the support team:
+    - releasing new code
+    - environment access
+    - database tasks
+    - secret distribution
```

Comment:
I don't think containerization on its own helps with any of these.

Reply:
This is fair, I noticed that I struggled to find the words to explain myself. 

I've had another crack at it to try and explain that deciding to be consistent on containers may then allow for consistent processes to develop. 

Without consistency in Containers we are going to struggle to make our tooling and processes for common stuff consistent too. With consistent processes that should be usable across container platforms (apart from Heroku) we solve the problem of having to learn and re learn how each project does things.